include/deal.II/matrix_free/dof_info.templates.h

```diff
@@ -183,6 +183,98 @@ namespace internal
 
 
 
+    void
+    DoFInfo::get_dof_indices(std::vector<unsigned int> &my_rows,
+                             const unsigned int         cell,
+                             const bool                 apply_constraints) const
+    {
+      const unsigned int n_fe_components = start_components.back();
+      const unsigned int fe_index =
+        dofs_per_cell.size() == 1 ? 0 : cell_active_fe_index[cell];
+      const unsigned int dofs_this_cell = dofs_per_cell[fe_index];
+
+      const unsigned int n_vectorization  = vectorization_length;
+      constexpr auto     dof_access_index = dof_access_cell;
+      const unsigned int n_vectorization_actual =
+        n_vectorization_lanes_filled[dof_access_index][cell];
+
+      if (index_storage_variants[dof_access_index][cell] ==
+          IndexStorageVariants::interleaved)
+        {
+          Assert(n_vectorization_actual == n_vectorization, ExcInternalError());
+          // in this case there are NO constraints on the cell so we
+          // know the number of indices we will read
+          my_rows.resize(n_vectorization_actual * dofs_this_cell);
+
+          const unsigned int *begin =
+            dof_indices_interleaved.data() +
+            row_starts[cell * n_vectorization * n_fe_components].first;
+
+          const unsigned int *end =
+            dof_indices_interleaved.data() +
+            row_starts[(cell + 1) * n_vectorization * n_fe_components].first;
+
+          Assert(static_cast<unsigned int>(end - begin) == my_rows.size(),
+                 ExcInternalError());
+          std::copy(begin, end, my_rows.begin());
```

Comment:
Is there a particular reason you add this block? The block below should also have all data filled, so you can simply use the `else` case every time? The order of the DoFs will be different but we do not really care about that anyway? In any case, the non-interleaved variant below is more natural because it lists the numbers cell by cell, rather than in the interleaved format here.

Reply:
> The block below should also have all data filled, so you can simply use the else case every time?

that was not obvious to me from the code in FEEvalution 😄 , so I guess I just followed the logic there. I guess the only minor connivence is that in this case we can read all DoFs at once and we know the size a-priori (so single resize). Let's see if nothing gets broken by always going through the `else` branch....